src/features/projectsV2/ProjectsMap/microComponents/PlantLocations.tsx

```diff
@@ -0,0 +1,232 @@
+
+  const getDateDiff = (pl: PlantLocation) => {
+    if (!pl.interventionStartDate) {
+      return null;
+    }
+    const today = new Date();
+    const plantationDate = new Date(pl.interventionStartDate?.slice(0, 10));
+    const differenceInTime = today.getTime() - plantationDate.getTime();
+    const differenceInDays = differenceInTime / (1000 * 3600 * 24);
+    if (differenceInDays < 1) {
+      return t('today');
+    } else if (differenceInDays < 2) {
+      return t('yesterday');
+    } else if (differenceInDays <= 10) {
+      return t('daysAgo', {
+        days: localizedAbbreviatedNumber(locale, differenceInDays, 0),
+      });
+    } else {
+      return null;
+    }
+  };
+
+  const makeInterventionGeoJson = (
+    geometry: Point | Polygon,
+    id: string,
+    extra?: Record<string, string | number | boolean | null>
+  ): Feature<Point | Polygon> => {
+    const properties = {
+      id,
+      ...extra,
+    };
+
+    return {
+      type: 'Feature',
+      properties,
+      geometry,
+    };
+  };
+
+  const features = plantLocations.map((el) => {
+    const isSelected =
+      selectedPlantLocation && selectedPlantLocation.id === el.id;
+    const isHovered = hoveredPlantLocation && hoveredPlantLocation.id === el.id;
+    const GeoJSON = makeInterventionGeoJson(el.geometry, el.id, {
+      highlightLine: isSelected || isHovered,
+      opacity:
+        el.type === 'multi-tree-registration' ? getPolygonColor(el) : 0.5,
+      dateDiff: getDateDiff(el),
+    });
+    return GeoJSON;
+  });
+  return (
+    <>
+      <Source
+        id={'display-source'}
+        type="geojson"
+        data={{
+          type: 'FeatureCollection',
+          features: [...features],
+        }}
+      >
+        <Layer
+          id={`plant-polygon-layer`}
+          type="fill"
+          paint={{
+            'fill-color': isSatelliteView ? '#ffffff' : '#007A49',
+            'fill-opacity': ['get', 'opacity'],
+          }}
+          filter={['==', ['geometry-type'], 'Polygon']}
+        />
+        <Layer
+          id={`point-layer`}
+          type="circle"
+          paint={{
+            'circle-color': isSatelliteView ? '#ffffff' : '#007A49',
+            'circle-opacity': 0.5,
+          }}
+          filter={['==', ['geometry-type'], 'Point']}
+        />
+        <Layer
+          id={`line-selected`}
+          type="line"
+          paint={{
+            'line-color': isSatelliteView ? '#ffffff' : '#007A49',
+            'line-width': 4,
+          }}
+          filter={['==', ['get', 'highlightLine'], true]}
+        />
+        <Layer
+          id={`datediff-label`}
+          type="symbol"
+          layout={{
+            'text-field': ['get', 'dateDiff'],
+            'text-anchor': 'center',
+            'text-font': ['Ubuntu Regular'],
+          }}
+          paint={{
+            'text-color': isSatelliteView ? '#ffffff' : '#2f3336',
+          }}
+          filter={['!=', ['get', 'dateDiff'], null]}
+        />
```

Comment:
Resolve the warning seen for the `filter` expression

`Type 'null' is not assignable to type 'ExpressionSpecification | ExpressionInputType'.`

Reply:
I still see the warning.

Reply:
resolved